crates/proc_macro_api/src/msg.rs

```diff
@@ -77,13 +77,23 @@ impl Message for Request {}
 impl Message for Response {}
 
 fn read_json(inp: &mut impl BufRead) -> io::Result<Option<String>> {
-    let mut buf = String::new();
-    inp.read_line(&mut buf)?;
-    buf.pop(); // Remove trailing '\n'
-    Ok(match buf.len() {
-        0 => None,
-        _ => Some(buf),
-    })
+    loop {
+        let mut buf = String::new();
+        inp.read_line(&mut buf)?;
+        buf.pop(); // Remove trailing '\n'
+
+        // Some ill behaved macro try to use stdout for debugging
+        // We ignore it here
+        if !buf.starts_with("{") {
+            log::error!("proc-macro try to print : {}", buf);
+            continue;
+        }
+
+        return Ok(match buf.len() {
+            0 => None,
+            _ => Some(buf),
+        });
+    }
```

Comment:
I don't know a forever loop is the best option here, but since any error inside proc-macro should be return in  `inp.read_line?`. So I think it is okay to do it, but let someone reviews it first to confirm my idea :)

Reply:
I think the loop is fine, but we should move `if len == 0 { return None }` check before `starts_with("{")` check, otherwise we'll loop forever in some cases.

Reply:
Oops... Fixed.